### PR TITLE
Add GitHub Actions Workflows

### PR DESCRIPTION
**Pull request by ajschmidt8**
## Description

This PR adds GitHub Actions workflows to `cuspatial`.

### Task list

Coverage required for this PR:
- [x] Python tests
- [x] Style checks
- [x] Notebooks

Future work:
- JUnit output annotations? https://github.com/marketplace/actions/junit-report-action
- Reduce exit code handling verbosity (use global variable with `trap` function)
- Set up `rapids-dependency-file-generator` pre-commit hook
- Docs builds
- Codecov? (it doesn't seem like `codecov` was ever set up for `cuxfilter`)

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cuspatial/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: conda/recipes/cuxfilter/meta.yaml [hunk 1]
```diff
@@ -1,9 +1,9 @@
 # Copyright (c) 2018-2020, NVIDIA CORPORATION.
 
 {% set version = environ.get('GIT_DESCRIBE_TAG', '0.0.0.dev').lstrip('v') + environ.get('VERSION_SUFFIX', '') %}
-{% set minor_version =  version.split('.')[0] + '.' + version.split('.')[1] %}
+{% set minor_version = version.split('.')[0] + '.' + version.split('.')[1] %}
 {% set py_version=environ.get('CONDA_PY', 36) %}
```
Comment (bdice) [inline conda/recipes/cuxfilter/meta.yaml:5]:
```suggestion
{% set py_version=environ.get('CONDA_PY', 38) %}
```

File: ci/test_python.sh [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+set +e
+
+# TODO: exit code handling is too verbose. Find a cleaner solution.
+rapids-logger "pytest cuxfilter"
+pushd python/
+pytest \
+  --cache-clear \
+  --junitxml="${RAPIDS_TESTS_DIR}/junit-cuxfilter.xml" \
+  --cov-config=.coveragerc \
```
Comment (bdice) [inline ci/test_python.sh:46]: This needs `pytest-cov` in `dependencies.yaml` to fix the current  CI failure.

File: ci/check_style.sh [hunk 3]
```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
```
Comment (bdice) [inline ci/check_style.sh:1]: @ajschmidt8 This file can be dramatically simplified now that #420 has been merged.

**Comment by ajschmidt8** [comment]
rerun tests

**Comment by ajschmidt8** [comment]
rerun tests

**Review by bdice — APPROVED** [review]
All looks good, just need one small tweak.

File: .github/workflows/pr.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  pr-builder:
+    needs:
+      - checks
+      - conda-python-build
+      - conda-python-tests
```
[the remark this answers is not in the document]
Comment (ajschmidt8) [reply .github/workflows/pr.yaml:17]: thanks! resolved in 065f910.

**Review by AjayThorve — APPROVED** [review]

File: ci/test_notebooks.sh [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+conda activate test
+set -u
+
+rapids-print-env
+
+rapids-logger "Downloading artifacts from previous jobs"
+PYTHON_CHANNEL=$(rapids-download-conda-from-s3 python)
+
+rapids-mamba-retry install \
+  -c "${PYTHON_CHANNEL}" \
```
Comment (ajschmidt8) [inline ci/test_notebooks.sh:27]:
```suggestion
  --channel "${PYTHON_CHANNEL}" \
```

File: ci/test_python.sh [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+RAPIDS_TESTS_DIR=${RAPIDS_TESTS_DIR:-"${PWD}/test-results"}
+RAPIDS_COVERAGE_DIR=${RAPIDS_COVERAGE_DIR:-"${PWD}/coverage-results"}
+mkdir -p "${RAPIDS_TESTS_DIR}" "${RAPIDS_COVERAGE_DIR}"
+SUITEERROR=0
+
+rapids-print-env
+
+rapids-mamba-retry install \
+  -c "${PYTHON_CHANNEL}" \
```
Comment (ajschmidt8) [inline ci/test_python.sh:32]:
```suggestion
  --channel "${PYTHON_CHANNEL}" \
```

**Pull request merged**